src/sage/combinat/k_regular_sequence.py

```diff
@@ -918,6 +979,367 @@ def _n_to_index_(self, n):
+            sage: S
+            2-regular sequence 1, 2, 2, 4, 2, 4, 4, 8, 2, 4, ...
+            sage: from itertools import islice
+            sage: L = []; ps = 0
+            sage: for j in islice(S, 110):
+            ....:     ps += j
+            ....:     L.append(ps)
+            sage: G = Seq2.guess(lambda n: L[n])
+            sage: G
+            2-regular sequence 1, 3, 5, 9, 11, 15, 19, 27, 29, 33, ...
+            sage: G.linear_representation()
+            ((1, 0),
+             Finite family {0: [ 0  1]
+                               [-3  4],
+                            1: [3 0]
+                               [3 2]},
+             (1, 1))
+            sage: G == S.partial_sums(include_n=True)
+            True
+
+        ::
+
+            sage: Seq3 = kRegularSequenceSpace(3, QQ)
+            sage: S = Seq3((Matrix([1]), Matrix([3]), Matrix([2])), vector([1]), vector([1]))
+            sage: S
+            3-regular sequence 1, 3, 2, 3, 9, 6, 2, 6, 4, 3, ...
+            sage: from itertools import islice
+            sage: L = []; ps = 0
+            sage: for j in islice(S, 110):
+            ....:     ps += j
+            ....:     L.append(ps)
+            sage: G = Seq3.guess(lambda n: L[n])
+            sage: G
+            3-regular sequence 1, 4, 6, 9, 18, 24, 26, 32, 36, 39, ...
+            sage: G.linear_representation()
+            ((1, 0),
+             Finite family {0: [ 0  1]
+                               [-6  7],
+                            1: [18/5  2/5]
+                               [18/5 27/5],
+                            2: [ 6  0]
+                               [24  2]},
+             (1, 1))
+            sage: G == S.partial_sums(include_n=True)
+            True
+
+        ::
+
+            sage: Seq2.guess(s, max_exponent=1)
+            Traceback (most recent call last):
+            ...
+            RuntimeError: aborting as exponents would be larger than max_exponent=1
+        """
+        import logging
+        logger = logging.getLogger(__name__)
+
+        from sage.arith.srange import srange, xsrange
+        from sage.matrix.constructor import Matrix
+        from sage.misc.mrange import cantor_product
+        from sage.modules.free_module_element import vector
+
+        k = self.k
+        domain = self.coefficient_ring()
+        if sequence is None:
+            mu = [[] for _ in srange(k)]
+            seq = lambda m: vector([])
+        else:
+            mu = [M.rows() for M in sequence.mu]
+            seq = lambda m: sequence.left * sequence._mu_of_word_(
+                self._n_to_index_(m))
+            logger.info('including %s', sequence)
+
+        zero = domain(0)
+        one = domain(1)
+
+        # A `line` will be a triple `(t, r, s)` corresponding to an entry
+        # `k**t * m + r` belonging to `M_s`
+        # TODO: what is `M_s`?
+
+        # The elements of `lines` will be correspond to the current components
```

Comment:
```suggestion
        # The elements of `lines` will correspond to the current components
```
https://github.com/dkrenn/sage/pull/1#discussion_r1207075106